src/accelerate/state.py

```diff
@@ -179,7 +179,10 @@ def __init__(self, cpu: bool = False, **kwargs):
                 )
 
             # Sets up self.backend + imports
-            backend, distributed_type = self._prepare_backend(cpu, use_sagemaker_dp, kwargs.pop("backend", None))
+            original_backend = kwargs.pop("backend", None)
+            backend, distributed_type = self._prepare_backend(cpu, use_sagemaker_dp, original_backend)
+            if original_backend and backend != original_backend:
+                logger.warning(f"The assigned backend is {original_backend}, but the real backend is {backend}")
```

Comment:
I don't think it makes sense to do a warning here. This should be a `raise` if this is truly a situation we can get into.

Reply:
Done

Reply:
I don't see this change propagated?

Reply:
Sorry, I forgot to push. It should be updated now, pls check. Thx.